roscpp_azure_iothub/src/ros_azure_iothub_cpp_node.cpp

```diff
@@ -244,25 +292,56 @@ void topicCallback(const topic_tools::ShapeShifter::ConstPtr& msg,
     parser.deserializeIntoFlatContainer( topic_name, Span<uint8_t>(buffer), &flat_container, 100);
     parser.applyNameTransform( topic_name, flat_container, &renamed_values );
 
-    ROS_INFO("--------- %s ----------", topic_name.c_str() );
-    sendMsgToAzureIoTHub(topic_name.c_str(), deviceHandle);
-    for (auto it: renamed_values)
+    // Send info to IoTHub via reported properties 
+    if (!(std::find(topicsToReport.begin(), topicsToReport.end(), topic_name.c_str()) == topicsToReport.end()))
     {
-        const std::string& key = it.first;
-        const Variant& value   = it.second;
-        char _buffer [256] = {0};
-        snprintf(_buffer, sizeof(_buffer), " %s = %f", key.c_str(), value.convert<double>());
-        ROS_INFO("%s",_buffer);
-        sendMsgToAzureIoTHub(_buffer, deviceHandle);
+        ROS_INFO("Sending message from %s to IoTHub via reported properties ", topic_name.c_str()); 
+
+        std::string topic_msg = "";
+        bool first = true;
+
+        for (auto it: renamed_values)
+        {
+            const std::string& key = it.first;
+            const Variant& value   = it.second;
+            char _buffer [256] = {0};
```

Comment:
why 256?
underscores are typically only used for member variables - this is likely a hold over; feel free to remove it for this local variable.

Reply:
For the 256, I based it off of the telemetry buffer since similar messages would be sent. For the _buffer variable, there seems to be a buffer variable in the parent scope, but I can rename it something different.

Reply:
snprintf will return the count of characters required; we should handle the case when there is an overflow so we don't drop values. Let's create a bug for this and fix it in another checkin.